### PR TITLE
v2: Changes in Table of Content

### LINKED ISSUE #1213: PDF: Changes in Table of Content
The new specifications introduces a third possible legal status (running modifications ?with anticipated effects?) on top of 'in force' and 'running modifications'. 
As soon as a real estate is concerned by any of these three cases, the topic has to be listed in the concerned topic list. If for a given topic only one or two of the law status concern the real estate, the status is omitted in the concerned topic list of the TOC but NOT displayed in the not concerned topic list (see also section 3.3.1 from the link below).

Details on pages 12 and 23 of the specification: https://github.com/openoereb/pyramid_oereb/files/6442522/Weisung-statischer-Auszug-de.pdf

### What needs to be done:
- [x] Change display logic of topics regarding their legal status
- [x] Move update date of the surveying data to the title page if not already done in #1212
- [x] Include information on restrictions potentially referenced in the land registry
- [x] Verify layout conformance (spacing, padding, dimensions, font sizes)
- [ ] Check for renamed attributes (TO DO: rename "exclusion of liability" to "disclaimer" when https://github.com/openoereb/pyramid_oereb/pull/1317/ is merged)
- [x] QR-Code -> see dedicated ticket #1270

### PR DESCRIPTION
**Pull request by vuilleumierc**
Closes https://github.com/openoereb/pyramid_oereb/issues/1213
Goes with https://github.com/openoereb/pyramid_oereb/pull/1325

### CODE REVIEW

File: print-apps/oereb/general_info_and_disclaimer.jrxml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+					<property name="com.jaspersoft.studio.unit.y" value="mm"/>
+				</reportElement>
+				<textElement markup="styled">
+					<font fontName="Cadastra" size="6" isBold="true"/>
+					<paragraph lineSpacingSize="8.0"/>
+				</textElement>
+				<textFieldExpression><![CDATA[$P{GeneralInformationLabel}]]></textFieldExpression>
+			</textField>
+		</band>
+	</title>
```
Comment (marionb) [inline print-apps/oereb/general_info_and_disclaimer.jrxml:38]: I think I would have used detail bands here. But if this is working I would not touch it ;-)
If something works in the Print templates it is best to not touch it :-D

**Review by marionb — APPROVED** [review]
thanks for the work and the changes. LGTM!

File: print-apps/oereb/config.yaml [hunk 2]
```diff
@@ -161,11 +161,10 @@ templates:
...
                             OfficeAtWeb: *optStr
-            ExclusionOfLiability: !datasource
-                default: []
-                attributes:
-                    Title: !string {} #/definitions/MultilingualText
-                    Content: !string {} #/definitions/MultilingualText
+            DisclaimerGrundbuch_Title: !string {}
+            DisclaimerGrundbuch_Content: !string {}
+            DisclaimerKbS_Title: !string {}
+            DisclaimerKbS_Content: !string {}
```
Comment (voisardf) [inline print-apps/oereb/config.yaml]: If you change the attribute names like I suggested, then you'll need: 
            DisclaimerLandRegister_Title: !string {}
            DisclaimerLandRegister_Content: !string {}
            DisclaimerPollutedSites_Title: !string {}
            DisclaimerPollutedSites_Content: !string {}

🙂

File: print-apps/oereb/general_info_and_disclaimer.jrxml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Created with Jaspersoft Studio version 6.17.0.final using JasperReports Library version 6.17.0-6d93193241dd8cc42629e188b94f9e0bc5722efd  -->
+<jasperReport xmlns="http://jasperreports.sourceforge.net/jasperreports" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://jasperreports.sourceforge.net/jasperreports http://jasperreports.sourceforge.net/xsd/jasperreport.xsd" name="topicmapboxlegend" pageWidth="252" pageHeight="155" whenNoDataType="AllSectionsNoDetail" columnWidth="252" leftMargin="0" rightMargin="0" topMargin="0" bottomMargin="0" uuid="6477891d-f5c5-456f-a115-e91b2a2768c6">
+	<property name="com.jaspersoft.studio.data.defaultdataadapter" value="One Empty Record"/>
+	<property name="com.jaspersoft.studio.unit." value="mm"/>
+	<style name="Default" isDefault="true" fontName="Cadastra" fontSize="8"/>
+	<parameter name="GeneralInformation" class="java.lang.String"/>
+	<parameter name="GeneralInformationLabel" class="java.lang.String"/>
+	<parameter name="DisclaimerGrundbuch_Title" class="java.lang.String"/>
```
Comment (voisardf) [inline print-apps/oereb/general_info_and_disclaimer.jrxml]: same here:
```
<parameter name="DisclaimerLandRegister_Title" class="java.lang.String"/>
<parameter name="DisclaimerLandRegister_Content" class="java.lang.String"/>
```

File: print-apps/oereb/general_info_and_disclaimer.jrxml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+				<reportElement x="0" y="6" width="230" height="11" uuid="b1952a7b-2006-4a21-bf98-8162489e3bac">
+					<property name="com.jaspersoft.studio.unit.width" value="mm"/>
+					<property name="com.jaspersoft.studio.unit.x" value="mm"/>
+					<property name="com.jaspersoft.studio.unit.y" value="mm"/>
+					<property name="com.jaspersoft.studio.unit.height" value="mm"/>
+				</reportElement>
+				<textElement>
+					<font fontName="Cadastra" size="6" isBold="true"/>
+				</textElement>
+				<textFieldExpression><![CDATA[$P{DisclaimerGrundbuch_Title}]]></textFieldExpression>
```
Comment (voisardf) [inline print-apps/oereb/general_info_and_disclaimer.jrxml]: and here:
`<textFieldExpression><![CDATA[$P{DisclaimerLandRegister_Title}]]></textFieldExpression>`

File: print-apps/oereb/general_info_and_disclaimer.jrxml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+			</textField>
+			<textField textAdjust="StretchHeight" isBlankWhenNull="true">
+				<reportElement positionType="Float" stretchType="ContainerBottom" x="0" y="17" width="230" height="38" isPrintWhenDetailOverflows="true" uuid="688222aa-3250-4873-95e8-a6485dce6c9f">
+					<property name="com.jaspersoft.studio.unit.height" value="mm"/>
+					<property name="com.jaspersoft.studio.unit.y" value="mm"/>
+				</reportElement>
+				<textElement>
+					<font fontName="Cadastra" size="6"/>
+				</textElement>
+				<textFieldExpression><![CDATA[$P{DisclaimerGrundbuch_Content}]]></textFieldExpression>
```
Comment (voisardf) [inline print-apps/oereb/general_info_and_disclaimer.jrxml]: `<textFieldExpression><![CDATA[$P{DisclaimerLandRegister_Content}]]></textFieldExpression>`

File: print-apps/oereb/pdfextract.jrxml [hunk 6]
```diff
@@ -36,7 +36,10 @@
 	<parameter name="northArrowSubReport" class="java.lang.String"/>
 	<parameter name="scalebarSubReport" class="java.lang.String"/>
 	<parameter name="GeneralInformation" class="java.lang.String"/>
-	<parameter name="ExclusionOfLiabilityDataSource" class="net.sf.jasperreports.engine.JRDataSource"/>
+	<parameter name="DisclaimerGrundbuch_Title" class="java.lang.String"/>
```
Comment (voisardf) [inline print-apps/oereb/pdfextract.jrxml]:
```
	<parameter name="DisclaimerLandRegister_Title" class="java.lang.String"/>
	<parameter name="DisclaimerLandRegister_Content" class="java.lang.String"/>
	<parameter name="DisclaimerPollutedSites_Title" class="java.lang.String"/>
	<parameter name="DisclaimerPollutedSites_Content" class="java.lang.String"/>
```

File: print-apps/oereb/pdfextract.jrxml [hunk 7]
```diff
@@ -247,12 +250,21 @@
 					<subreportParameter name="ThemeWithoutDataDataSource">
 						<subreportParameterExpression><![CDATA[$P{ThemeWithoutDataDataSource}]]></subreportParameterExpression>
 					</subreportParameter>
-					<subreportParameter name="ExclusionOfLiabilityDataSource">
-						<subreportParameterExpression><![CDATA[$P{ExclusionOfLiabilityDataSource}]]></subreportParameterExpression>
-					</subreportParameter>
 					<subreportParameter name="GeneralInformation">
 						<subreportParameterExpression><![CDATA[$P{GeneralInformation}]]></subreportParameterExpression>
 					</subreportParameter>
+					<subreportParameter name="DisclaimerGrundbuch_Title">
```
Comment (voisardf) [inline print-apps/oereb/pdfextract.jrxml]:
```
					<subreportParameter name="DisclaimerLandRegister_Title">
						<subreportParameterExpression><![CDATA[$P{DisclaimerLandRegister_Title}]]></subreportParameterExpression>
					</subreportParameter>
					<subreportParameter name="DisclaimerLandRegister_Content">
						<subreportParameterExpression><![CDATA[$P{DisclaimerLandRegister_Content}]]></subreportParameterExpression>
					</subreportParameter>
					<subreportParameter name="DisclaimerPollutedSites_Title">
						<subreportParameterExpression><![CDATA[$P{DisclaimerPollutedSites_Title}]]></subreportParameterExpression>
					</subreportParameter>
					<subreportParameter name="DisclaimerPollutedSites_Content">
						<subreportParameterExpression><![CDATA[$P{DisclaimerPollutedSites_Content}]]></subreportParameterExpression>
					</subreportParameter>
```

File: print-apps/oereb/toc.jrxml [hunk 8]
```diff
@@ -63,7 +63,10 @@
 	<parameter name="PLRCadastreAuthority_City" class="java.lang.String"/>
 	<parameter name="Signature" class="java.lang.String"/>
 	<parameter name="GeneralInformation" class="java.lang.String"/>
-	<parameter name="ExclusionOfLiabilityDataSource" class="net.sf.jasperreports.engine.JRDataSource"/>
+	<parameter name="DisclaimerGrundbuch_Title" class="java.lang.String"/>
```
Comment (voisardf) [inline print-apps/oereb/toc.jrxml]:
```
	<parameter name="DisclaimerLandRegister_Title" class="java.lang.String"/>
	<parameter name="DisclaimerLandRegister_Content" class="java.lang.String"/>
	<parameter name="DisclaimerPollutedSites_Title" class="java.lang.String"/>
	<parameter name="DisclaimerPollutedSites_Content" class="java.lang.String"/>
```

File: print-apps/oereb/toc.jrxml [hunk 9]
```diff
@@ -254,37 +258,60 @@
...
+					<topPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+					<leftPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+					<bottomPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+					<rightPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+				</box>
+				<textElement markup="html">
+					<font size="6" isBold="true"/>
 				</textElement>
-				<textFieldExpression><![CDATA[$P{GeneralInformation}]]></textFieldExpression>
+				<textFieldExpression><![CDATA[$P{DisclaimerKbS_Title}]]></textFieldExpression>
```
Comment (voisardf) [inline print-apps/oereb/toc.jrxml]: `<textFieldExpression><![CDATA[$P{DisclaimerPollutedSites_Title}]]></textFieldExpression>`

File: print-apps/oereb/toc.jrxml [hunk 10]
```diff
@@ -254,37 +258,60 @@
...
+					<topPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+					<leftPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+					<bottomPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+					<rightPen lineWidth="0.0" lineStyle="Solid" lineColor="#000000"/>
+				</box>
+				<textElement textAlignment="Left" markup="html">
+					<font size="6" isBold="false"/>
 				</textElement>
-				<textFieldExpression><![CDATA[$R{GeneralInformationLabel}]]></textFieldExpression>
+				<textFieldExpression><![CDATA[$P{DisclaimerKbS_Content}]]></textFieldExpression>
```
Comment (voisardf) [inline print-apps/oereb/toc.jrxml]: `<textFieldExpression><![CDATA[$P{DisclaimerPollutedSites_Content}]]></textFieldExpression>`

File: print-apps/oereb/toc.jrxml [hunk 11]
```diff
@@ -254,37 +258,60 @@
...
+					<property name="com.jaspersoft.studio.unit.width" value="mm"/>
+					<property name="com.jaspersoft.studio.unit.height" value="mm"/>
+				</reportElement>
+				<subreportParameter name="GeneralInformation">
+					<subreportParameterExpression><![CDATA[$P{GeneralInformation}]]></subreportParameterExpression>
+				</subreportParameter>
+				<subreportParameter name="GeneralInformationLabel">
+					<subreportParameterExpression><![CDATA[$R{GeneralInformationLabel}]]></subreportParameterExpression>
+				</subreportParameter>
+				<subreportParameter name="DisclaimerGrundbuch_Title">
```
Comment (voisardf) [inline print-apps/oereb/toc.jrxml]: and last:
```

				<subreportParameter name="DisclaimerLandRegister_Title">
					<subreportParameterExpression><![CDATA[$P{DisclaimerLandRegister_Title}]]></subreportParameterExpression>
				</subreportParameter>
				<subreportParameter name="DisclaimerLandRegister_Content">
					<subreportParameterExpression><![CDATA[$P{DisclaimerLandRegister_Content}]]></subreportParameterExpression>
				</subreportParameter>
```

**Review by voisardf — APPROVED** [review]
looks fine, thanks. Same suggested changes, if you accept them.

**Comment by vuilleumierc** [comment]
Thanks @voisardf, the changes are implemented here as well

**Pull request merged**